operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml

```diff
@@ -588,14 +588,127 @@ spec:
             mountPath: /etc/prometheus
           - mountPath: /etc/istio-certs
             name: istio-certs
+        - name: istio-proxy
+          image: "gcr.io/istio-testing/proxyv2:latest"
+          ports:
+            - containerPort: 15090
+              protocol: TCP
+              name: http-envoy-prom
+          args:
+            - proxy
```

Comment:
I am really not sure it is appropriate to spin up an envoy that connects to pilot, sds, etc etc just to dump a cert to a file... I know it makes the cert rotation a bit easier since envoy handles it, but it seems way overkill to me

Reply:
I think this makes sure we don't need to build a separate sidecar containing only the istio-agent, also the sidecar can be used for other purposes.

Reply:
I don't care about the envoy binary existing in the container, but we shouldn't run it

Reply:
I'm fine with not running Envoy. @costinm WDYT?

Reply:
Confirmed this is not a P0. Let's keep it running for now. The overhead is pretty minimal.